src/helics/core/mpi/MpiComms.cpp

```diff
@@ -9,38 +9,319 @@ Lawrence Livermore National Laboratory, operated by Lawrence Livermore National
 
 */
 #include "MpiComms.h"
-#include "helics/core/ActionMessage.h"
+#include "../../common/AsioServiceManager.h"
+#include "../ActionMessage.hpp"
 #include <memory>
 
+#include <mpi.h>
+
 namespace helics
 {
-MpiComms::MpiComms (const std::string &brokerTarget, const std::string &localTarget)
-    : CommsInterface (brokerTarget, localTarget)
+
+bool MpiComms::mpiCommsExists = false;
```

Comment:
If you want this to actually work it probably needs to be atomic otherwise it isn't going to work in multithreaded cases